packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -438,10 +440,25 @@ class WidgetTester extends WidgetController implements HitTestDispatcher, Ticker
     return TestAsyncUtils.guard<void>(() {
       binding.attachRootWidget(widget);
       binding.scheduleFrame();
+      if (_firstPumpWidget) {
+        _firstPumpWidget = false;
+        // Emulate what `RenderObjectToWidgetAdapter.attachToRenderTree` does
+        // when it builds a new tree from scratch.
+        binding.buildOwner.buildScope(binding.renderViewElement);
```

Comment:
We should have a test that verifies this behavior specifically, namely that we build the widgets without kicking off layout.

Reply:
Good idea.

Reply:
what does this actually do, meaningfully? should we be changing how runApp works?

Reply:
It emulates

https://github.com/flutter/flutter/blob/fabf4e3d0d311181178d2c601d29a2f739ea543a/packages/flutter/lib/src/widgets/binding.dart#L967

The only useful thing I see in the latter is that it mounts the element as the root of the scene. I wonder if we can call `element.mount` without `owner.buildScope`. What does `buildScope` do for lazy-built elements if there's no layout information?